test/metric_value_benchmark/prometheus_test.go

```diff
@@ -99,12 +103,58 @@ func (t *PrometheusTestRunner) validatePrometheusMetric(metricName string) statu
 		Status: status.FAILED,
 	}
 
-	fetcher, err := t.MetricFetcherFactory.GetMetricFetcher(metricName)
-	if err != nil {
+	dims, failed := t.Base.DimensionFactory.GetDimensions([]dimension.Instruction{})
```

Comment:
yeah probably. i was trying to make dims and failed accessible from outside of switch but nils complain. I will see if I can do anything about it


also for the comment below: wow that didn't work when I tried so that's why I did this Base thing. dang.